src/main/java/nus/climods/model/ModelManager.java

```diff
@@ -123,6 +126,16 @@ public void setModuleInFocus(Module module) throws ApiException {
         setFilteredModuleList(new ViewModulePredicate(module.getCode()));
     }
 
+    @Override
+    public boolean showModules(List<String> moduleCodes) {
+        long validCodes = moduleCodes.stream().filter(this::isModuleOffered).count();
+        if (validCodes == 0) {
```

Comment:
This method helps to prevent filtering the list to empty if there were no valid module codes in the prerequisite string (e.g if all the codes are from a past year). The boolean is so we can show the right message to the user.